backend/app/Traits/UniqueCodeTrait.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace App\Traits;
+
+use App\Models\Account;
+
+ //generates unique code for the account number
+ trait UniqueCodeTrait {   
```

Comment:
my changes for this file

highlighting `bootUniqueCodeTrait`, "bootTraitName" to be not in conflict with other traits
```
trait UniqueCodeTrait
{
    private static function _generateUniqueCode()
    {
        do {
            $code = (string)random_int(10000000000, 99999999999);
        } while (Account::where("account_number", $code)->exists());

        return $code;
    }

    /**
     * Boot function from Laravel.
     */
    protected static function bootUniqueCodeTrait()
    {
        // parent::boot();
        static::creating(function ($model) {
            if (empty($model->{$model->getUniqueCodeKeyName()})) {
                $code = static::_generateUniqueCode();

                $model->{$model->getUniqueCodeKeyName()} = $code;
            }
        });
    }

    /**
     * trait function for generating unique code
     * use case: seeder
     */
    public function generateUniqueCode()
    {
        return static::_generateUniqueCode();
    }
}
```